### PR TITLE
Add tech debt issue template

### PR DESCRIPTION
**Pull request by jbohanon**
# Description

Add an issue template for tracking tech debt and code improvements akin to the one in Gloo Mesh Enterprise ([link](https://github.com/solo-io/gloo-mesh-enterprise/blob/main/.github/ISSUE_TEMPLATE/tech_debt.md?plain=1))

### CODE REVIEW

**Review by ashishb-solo — DISMISSED** [review]
looks good to me!

File: .github/ISSUE_TEMPLATE/TECH_DEBT.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    validations:
+      required: true
+  - type: input
+    id: version
+    attributes:
+      label: Gloo Edge Version
+      description: What version of your product are you using? Please specify a specific patch version
+      placeholder: eg v1.14.2
+    validations:
+      required: true
```
Comment (nfuden) [inline .github/ISSUE_TEMPLATE/TECH_DEBT.yml:22]: is this needed? I would figure most of the time we ould specify main

Comment (jbohanon) [reply .github/ISSUE_TEMPLATE/TECH_DEBT.yml:22]: Yeah I was thinking the same. Happy to remove it.

**Review by nfuden — APPROVED** [review]

**Review by ashishb-solo — APPROVED** [review]

**Pull request merged**